### PR TITLE
Fix misbehaviors when insert_distributed_one_random_shard = 1

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fix multiple block insertion into distributed table with `insert_distributed_one_random_shard = 1`. This is a marginal feature. Mark as improvement.

Detailed description / Documentation draft:

.

### CODE REVIEW

File: tests/queries/0_stateless/01615_random_one_shard_insertion.sql [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+drop table if exists shard_1.tbl;
 drop table if exists distr;
 
-create table shard (id Int32) engine = MergeTree order by cityHash64(id);
-create table distr as shard engine Distributed (test_cluster_two_shards_localhost, currentDatabase(), shard);
-
-insert into distr (id) values (0), (1);  -- { serverError 55; }
+create table shard_0.tbl (number UInt64) engine = MergeTree order by number;
+create table shard_1.tbl (number UInt64) engine = MergeTree order by number;
+create table distr (number UInt64) engine = Distributed(test_cluster_two_shards_different_databases, '', tbl);
```
Comment (azat) [inline tests/queries/0_stateless/01615_random_one_shard_insertion.sql:9]: If this test uses test_cluster_two_shards_different_databases cluster, then it is better to mark it as not suitable for parallel run (https://github.com/ClickHouse/ClickHouse/blob/82b6f49516c0cce17d28c6d70202f634492d830a/tests/queries/skip_list.json#L399)

**Review by azat — APPROVED** [review]

**Comment by azat** [comment]
>Fix multiple block insertion into distributed table with random_shard_insert = 1. This is a marginal feature. Mark as improvement.

For changelog it is better to use name of the setting `insert_distributed_one_random_shard`

**Comment by amosbird** [comment]
> > Fix multiple block insertion into distributed table with random_shard_insert = 1. This is a marginal feature. Mark as improvement.
> 
> For changelog it is better to use name of the setting `insert_distributed_one_random_shard`

Sure

File: tests/queries/0_stateless/01615_random_one_shard_insertion.sql (hunk 1 above)
Comment (amosbird) [reply tests/queries/0_stateless/01615_random_one_shard_insertion.sql:9]: done

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**